File: 天梯赛/L2-017. 人以群分.cpp
#include<iostream>
#include<vector>
#include<algorithm>
using namespace std;
int main(){
	int n,icnt,ocnt,sum=0,halfsum=0;
	cin>>n;
	vector<int> v(n);
	for(int i=0;i<n;++i){cin>>v[i];sum+=v[i];}
	sort(v.begin(),v.end());
	icnt=n/2;ocnt=n-icnt;
	for(int i=0;i<icnt;++i)halfsum+=v[i];
	printf("Outgoing #: %d\n",ocnt);
	printf("Introverted #: %d\n",icnt);
	printf("Diff = %d",sum-2*halfsum);
	return 0;
}

File: 天梯赛/java/L2-017. 人以群分.java
import java.util.Arrays;
import java.util.Scanner;

public class Main {
	
	public static void main(String[] args) {
		Scanner in=new Scanner(System.in);
		int n=in.nextInt(),sum=0,halfsum=0;
		int v[]=new int[n];
		for(int i=0;i<n;++i) {
			v[i]=in.nextInt();sum+=v[i];
		}
		Arrays.sort(v);
		int icnt=n/2,ocnt=n-icnt;
		for(int i=0;i<icnt;++i)halfsum+=v[i];
		System.out.printf("Outgoing #: %d\nIntroverted #: %d\nDiff = %d", ocnt,icnt,sum-2*halfsum);
;	}
	

}
